### PR TITLE
OpenAPI Docs Page Accessibility

### PR DESCRIPTION
**Pull request by jasonfrancis**
## Story
[STRY0013717](https://iu.service-now.com/nav_to.do?uri=rm_story.do?sys_id=c22dc83347a1bd909286dfbd436d435f%26sysparm_view=scrum) - API Documentation Accessibility Issues

## Context
A scan by the Digital Strategy Initiative found 3 issues to resolve on https://api.itpeople.iu.edu/ 
* Page language has not been identified
* Page missing headings *(They are objectively mistaken on this, maybe their scraper doesn't do javascript)*
* Skip to main content link is missing

All these items were able to be addressed by enriching the HTML string served by the `Documentation.RenderApiUI()` method.

## How Was This Tested?
This is the test.  Once it is deployed to the test environment I'll have them scrape it and see if the errors are remedied.

### CODE REVIEW

**Review by bmoberly — APPROVED** [review]

**Review by vbhupathi — APPROVED** [review]

**Comment by jasonfrancis** [comment]
Requesting a re-review since I've made substantial changes.

image


I've updated to use SiteImprove's fork of [Swagger-UI](https://github.com/swagger-api/swagger-ui).  So there are substantial visual changes at the [test API Documentation](https://api.itpeople-test.iu.edu).

Even using SiteImprove's version of Swagger-UI was not enough to pass a scan by SiteImprove's browser plugin.  So I had to add some javascript to massage the generated DOM into compliance.  I had to do a pretty bad hack using a `setTimeout()`to wait for more DOM to render. Much of the DOM is generated only when certain buttons are clicked, but I couldn't find any event handlers in Swagger-UI's docs to more correctly use.

If someone has a more elegant approach please recommend it.

**Review by bmoberly — APPROVED** [review]
I think this looks pretty good. It looks like tons of work was involved. I have three suggestions, but keep in mind that they are only suggestions if you feel that what we have is good enough:

1) remove the console.log statements when you feel that it's ok to do so
2) consider moving the script in Documentation.cs to its own file. This will make things easier to update in the future, as we'll get code completion, etc.
    * maybe for another story, but I'm sort of tempted to put the `APIUIMarkup` text in its own file and then load that file instead of a static string. But then again, if we refactor the script stuff into its own js file, the APIUIMarkup static string will be a lot shorter.
    * Again maybe for another story, but css can go in its own file too
3) it might be that we don't need the timer in the window.onload function. The [documentation that I'm finding](https://developer.mozilla.org/en-US/docs/Web/API/Window/load_event) says that the function gets called after everything is fully loaded in

**Review by bmoberly — APPROVED** [review]
I think this is good to go now if you're happy with it

File: src/API/Functions/Documentation.cs [hunk 1]
```diff
@@ -19,20 +19,185 @@ public static class Documentation
...
+			// End Swagger UI call region
+
+			window.ui = ui;
+		};
+
+		function fixDOM()
+		{
+			console.log(""Greetings from fixDOM()!"");
+
+			// Since we don't have good event hooks, wait X seconds for the DOM to render.
```
[the remark this answers is not in the document]
Comment (jasonfrancis) [reply src/API/Functions/Documentation.cs]: Thanks for the feedback.

Made the event handling more jquery-like, but stuck to vanilla JS in 9ab09a9.

Extracted the CSS and JS to their own files, and removed the extraneous `console.log()` calls in 599a261.

[the remark this answers is not in the document]
Comment (bmoberly) [reply src/API/Functions/Documentation.cs]: ty - resolved in https://github.com/indiana-university/itpeople-functions-v2/pull/109/commits/9ab09a9f64d00594eba3b7a304adf7e37231b50a and https://github.com/indiana-university/itpeople-functions-v2/commit/599a261d754c257866004fb7edc1cc939c157758

**Pull request merged**